File: CP/factorial.cpp
#include<bits/stdc++.h>
using namespace std;

int factorial(int n){
	if(n == 0 || n == 1)return 1;
	
	
	int smalloutput = factorial(n-1);
	
	return n*smalloutput;
}
int main(){
	int n;
	cin >> n;
	
	cout << factorial(n) << endl;
	
}

File: CP/factorial.java
import java.util.*;
public class factorial {
    public static int fact(int n){
        if(n==1)return 1;

        int smalloutput = fact(n-1);

        return n*smalloutput;
    }
    public static void main(String[] args){
        int ans = fact(5);
        System.out.println(ans);
    }
}
